spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/encryption/EncryptingConverter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.mongodb.core.encryption;
+
+import org.springframework.data.mongodb.core.convert.MongoConversionContext;
+import org.springframework.data.mongodb.core.convert.MongoValueConverter;
+
+/**
+ * A specialized {@link MongoValueConverter} for {@literal en-/decrypting} properties.
+ *
+ * @author Christoph Strobl
+ * @since 4.1
+ */
+public interface EncryptingConverter<S, T> extends MongoValueConverter<S, T> {
```

Comment:
valid point, we need to untangle bits here to avoid cycle between the `convert` and `encrypt` package. Basically we need to make sure `EncryptionContext` is not a `ValueConversionContext` any longer